### DISCUSSION TITLE
WPGraphQL Documents w/ Fragments

### DISCUSSION THREAD

**Discussion by archoda** [wp-graphql/wp-graphql#2979] [Ideas]
As I worked through utilizing the WPGraphQL Documents functionality I quickly found it cumbersome by not being able to reuse ( DRY ) fragments properly. It would be good if this functionality was in the documents featureset.

This could be done by saving a document and categorizing it strictly as a fragment. This fragment could then be tagged to be dynamically imported ( via a selector / picker or shortcode ) for N number of standard query document(s) for inclusion at runtime.

As a secondary item - it would also be great it the text/code kept it's formatting in the docs to maintain readability

**Comment by jasonbahl** [comment]
@archoda this is a great idea! We're working on re-building the GraphiQL IDE and this is a feature I think would fit well into that project. I opened a new issue on that repo to track / discuss this feature further: https://github.com/wp-graphql/wpgraphql-ide/issues/17
